src/middleware.ts

```diff
@@ -362,22 +371,12 @@ export const persist =
         }
       })
       .then(() => {
+        stateFromStorageInSync = { ...configResult, ...stateFromStorageInSync }
```

Comment:
```suggestion
        stateFromStorageInSync = stateFromStorageInSync && { ...configResult, ...stateFromStorageInSync }
```
We need this, right?

Reply:
![image](https://user-images.githubusercontent.com/35486736/123124253-ce123b80-d447-11eb-9958-1d004503cb65.png)
:wink:

Reply:
The point is creating a new object in the case of async: `configResult !== { ...configResult }`. In other words, `|| configResult` does never work.

Reply:
Yes it does. When using an async storage

Reply:
Oh, I see. In async, it doesn't reach this line before the return. Then, I'm fine with this.